front/src/modules/timesStops/helpers/scheduleData.ts

```diff
@@ -0,0 +1,35 @@
+import { ISO8601Duration2sec, datetime2sec, secToHoursString } from 'utils/timeManipulation';
+
+import type { ComputedScheduleEntry, ScheduleEntry } from '../types';
+
+/**
+ *
+ * @param schedule for a given operational point
+ * @param startTime time of departure from the very beginning of the route
+ */
+export function computeScheduleData(schedule: ScheduleEntry, startTime: string) {
+  if (!schedule) {
+    return { arrival: null, departure: null, stopFor: null };
+  }
+  const startTimeSeconds = datetime2sec(new Date(startTime));
+  // relative value, number of seconds since startTime
+  const arrivalSeconds = schedule.arrival
+    ? startTimeSeconds + ISO8601Duration2sec(schedule.arrival)
+    : null;
+  const stopForSeconds = schedule.stop_for ? ISO8601Duration2sec(schedule.stop_for) : null;
+  const departure =
+    arrivalSeconds && stopForSeconds ? startTimeSeconds + arrivalSeconds + stopForSeconds : null;
+  return {
+    arrival: arrivalSeconds,
+    departure,
+    stopFor: stopForSeconds,
```

Comment:
```suggestion
  const arrival = schedule.arrival
    ? startTimeSeconds + ISO8601Duration2sec(schedule.arrival)
    : null;
  const stopFor = schedule.stop_for ? ISO8601Duration2sec(schedule.stop_for) : null;
  const departure = arrival && stopFor ? startTimeSeconds + arrival + stopFor : null;
  return {
    arrival,
    departure,
    stopFor,
```

Reply:
it was maybe to explicitly show that there is seconds, but the function name seems enough

Reply:
I would rather keep it like this, because otherwise we don't know at first sight what we are manipulating. Is it ok for you ?